tests/src/test/scala/akka/kafka/scaladsl/IntegrationSpec.scala

```diff
@@ -183,25 +183,48 @@ class IntegrationSpec extends SpecBase with TestcontainersKafkaLike with Inside
       }
     }
 
-    "not produce any records after send-failure if stage is stopped" in {
+    "not produce or emit records downstream after send-failure and producer stage is stopped" in {
       assertAllStagesStopped {
+        val numMessagesBeforeFailure = 5
+        final class FailingStringSerializer extends org.apache.kafka.common.serialization.StringSerializer {
+          override def serialize(topic: String, data: String): Array[Byte] = {
+            if (data.toInt > numMessagesBeforeFailure)
+              throw new org.apache.kafka.common.errors.SerializationException()
+            else
+              super.serialize(topic, data)
+          }
+        }
+
         val topic1 = createTopic(1)
         val group1 = createGroupId(1)
-        // we use a 'max.block.ms' setting that will cause the metadata-retrieval to fail
-        // effectively failing the production of the first messages
-        val failFirstMessagesProducerSettings = producerDefaults.withProperty(ProducerConfig.MAX_BLOCK_MS_CONFIG, "1")
-
-        val producer = produce(topic1, 1 to 100, failFirstMessagesProducerSettings)
-        // assure the producer fails as expected
-        producer.failed.futureValue shouldBe a[org.apache.kafka.common.errors.TimeoutException]
-
-        val (control, probe) = createProbe(consumerDefaults.withGroupId(group1), topic1)
 
-        probe
+        val failFirstMessagesProducerSettings =
```

Comment:
This would fail the 6th message, not the first message, right?

Reply:
Yes, this was a leftover from the previous iteration of the test.  I'll update it if we decide to keep it.